core/src/main/java/org/apache/carbondata/core/metadata/SegmentFileStore.java

```diff
@@ -1133,17 +1133,23 @@ public static void deleteSegment(String tablePath, Segment segment,
     List<String> indexOrMergeFiles = fileStore.readIndexFiles(SegmentStatus.SUCCESS, true,
         FileFactory.getConfiguration());
     Map<String, List<String>> indexFilesMap = fileStore.getIndexFilesMap();
+    StringBuilder deletedFiles = new StringBuilder();
```

Comment:
do not use string builder, use java functional APIs like below

`String firstThenLast2 = files
	.stream()
	.map(p -> file.getPath)
	.collect(Collectors.toStringJoiner(","))
        .toString();`

try this

Reply:
used string.join with delimiter